WebApp/src/test/java/ev/projects/webapp/controllers/CaseControllerTest.java

```diff
@@ -1,93 +1,158 @@
+        TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username1, password1);
+        ResponseEntity<Optional<Case>> response = restTemplate.exchange(url, HttpMethod.GET,
+                null, new ParameterizedTypeReference<>() {});
+        Optional<Case> responseCase = response.getBody();
+        assertEquals(response.getStatusCode(), HttpStatus.OK);
         assertNotNull(responseCase);
         assertTrue(responseCase.isPresent());
-        Case rCase = responseCase.get();
-        assertEquals(aCase, rCase);
-        assertEquals(ID, rCase.getID());
-        assertEquals(title, rCase.getTitle());
-        assertEquals(creatorUser, rCase.getCreatorUser());
+        Case aCase = responseCase.get();
+        assertEquals( "English", aCase.getTitle());
+        assertEquals( "English lessons", aCase.getDescription());
+        assertEquals( caseCountOffset + 1, aCase.getID());
+        assertNotNull(aCase.getCreatorUser());
+        assertEquals( userCountOffset + 1, aCase.getCreatorUser().getID());
+        assertEquals(username1, aCase.getCreatorUser().getName());
+    }
+
+    @Test
+    @Order(1)
+    public void createCaseTest() {
+        String title = "History";
+        String description = "Legendary stories";
+        TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username1, password1);
+        HttpEntity<Case> createRequest = new HttpEntity<>(createCase(title, description, null));
+        ResponseEntity<Case> createResponse = restTemplate.exchange(baseUrl, HttpMethod.POST,
+                createRequest, new ParameterizedTypeReference<>() {});
+        assertEquals(HttpStatus.OK, createResponse.getStatusCode());
+        assertCase(restTemplate, title, description, username1, (int) (caseCountOffset + 2));
+    }
+
+    @Test
+    @Order(2)
+    public void updateCaseTest() {
+        String url = baseUrl + "/" + (caseCountOffset + 3) + "/";
+        String title = "Future";
+        String description = "Legendary tales";
+        TestRestTemplate restTemplate = new TestRestTemplate().withBasicAuth(username1, password1);
+        HttpEntity<Case> updateRequest = new HttpEntity<>(createCase(title, description, null));
+        ResponseEntity<Case> createResponse = restTemplate.exchange(url, HttpMethod.PUT,
+                updateRequest, new ParameterizedTypeReference<>() {});
+        assertEquals(createResponse.getStatusCode(), HttpStatus.OK);
+        assertCase(restTemplate, title, description, username1, (int) (caseCountOffset + 2));
+    }
+
+    private void assertCase(TestRestTemplate restTemplate, String title, String description, String username, int index) {
+        ResponseEntity<List<Case>> getResponse = restTemplate.exchange(baseUrl, HttpMethod.GET, null, new ParameterizedTypeReference<>() {});
+        List<Case> cases = getResponse.getBody();
+        assertEquals(getResponse.getStatusCode(), HttpStatus.OK);
```

Comment:
Too many asserts are hard to read. Consider adding additional helper method to compare fields instead of comparing every field in one method.

Reply:
I tried to split them this way: 8897048e7f4d9bb9d05ee38d5a438d0c14c39c18. However, I did not split assertions by few in one method because I don't know how to reasonably call the method then. For example if you split title and description assertion how you should call the method? I could call it assertCaseTitleAndDescription but that is kind of large to read.

Reply:
Could split assertions by methods with large names if you think that would be fine to read?

Reply:
Best practices says each test should ideally have one purpose - something like a use case. So if you for example want to assert that http status is ok and all fields are mapped correctly. You should have 2 different tests for each of these things. But since you doing more like a general functionality test, maybe it is ok... but just a thing to consider.